src/main/java/slimeknights/tconstruct/library/utils/ToolHelper.java

```diff
@@ -840,7 +840,11 @@ public static float calcCutoffDamage(float damage, float cutoff) {
   }
 
   public static float getActualDamage(ItemStack stack, EntityLivingBase player) {
-    float damage = (float) player.getEntityAttribute(SharedMonsterAttributes.ATTACK_DAMAGE).getAttributeValue();
+    float damage = (float) SharedMonsterAttributes.ATTACK_DAMAGE.getDefaultValue();
```

Comment:
Oh, I like this. Where does the call with null originate from?

Reply:
addInformation() called during building the search tree used by the creative tab on the client.

Reply:
addInformation() called during building the search tree used by the creative tab on the client.

Reply:
Shouldn't that have the local player? Or is that done before the server starts? Iirc the creative tabs are only called once the player is ingame.

Reply:
@bonii-xx They aren't, they're also called once in init. addInformation() can also be called on the server-side by any other mod.

Reply:
Probably something along those lines, meh.